### ISSUE TITLE
Update InfoMAP to a recent version

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2600]
Updating InfoMAP will likely be a breaking change, so it needs to happen before 1.0.

**Comment by ntamas** [comment]
Was there any progress regarding the licensing of InfoMAP? As far as I know it's still AGPL, isn't it?

Also, nothing stops us from adding an `igraph_community_infomap_new()` function instead of modifying the existing one, so we can still do this after 1.0. It does not have to block 1.0, especially considering our limited resources.

**Comment by szhorvat** [comment]
A reminder that we'll need to move from C++11 to C++14. This is fine but it must be noted in the changelog.

**Comment by ntamas** [comment]
The changelog of the `develop` branch now indicates that we now require C++14, FYI.

**Comment by vtraag** [comment]
> As far as I know it's still AGPL, isn't it?

No, they've accommodated our request and switched to GPL, see https://github.com/mapequation/infomap/commit/2bdcacf4e1f3c439db232f1a231105e081c32319.

**Comment by ntamas** [comment]
#2646 is merged inrto `develop` so closing.
